### PR TITLE
Plugins: Set migration handler to definne its own PanelMigrationModel and add tests

### PR DESCRIPTION
**Pull request by academo**
**What is this feature?**

Creates a separate type definition for `setMigrationHandler` inside the PanelPlugin that mirrors PanelModel's required properties.

**Why do we need this feature?**

Third-party code uses migration handlers but isn't validated against PanelModel changes. By creating a distinct type definition, we ensure breaking changes to PanelModel are caught at compile time when they affect migration handlers, rather than causing runtime failures in third-party code.

**Who is this feature for?**

All plugin code running in Grafana

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

**Review by oscarkilhed — APPROVED** [review]

File: packages/grafana-data/src/types/panel.ts [hunk 1]
```diff
@@ -134,12 +134,32 @@ export interface PanelEditorProps<T = any> {
   data?: PanelData;
 }
 
+/**
+ * This type mirrors the required properties from PanelModel<TOptions> needed for migration handlers.
```
[the remark this answers is not in the document]
Comment (academo) [reply packages/grafana-data/src/types/panel.ts:138]: @andresmgot that's one of the main reasons we are doing this, in reality not all of the attributes in the PanelModel (specially the optional) are passed to the setMigrationHandler. It is not a case of sometimes they are and sometimes they are not, but more than they never are available. So the PanelModel and what is passed to the migrationhandler are not the same.

[the remark this answers is not in the document]
Comment (andresmgot) [reply packages/grafana-data/src/types/panel.ts:138]: hmm, I am not sure I understand why (at least from the typings perspective). For example, the property `datasource` from PanelModel. It's defined [here](https://github.com/grafana/grafana/blob/academo/add-set-migration-handler-custom-type/public/app/features/dashboard/state/PanelModel.ts#L177) and set [here](https://github.com/grafana/grafana/blob/academo/add-set-migration-handler-custom-type/public/app/features/dashboard/state/PanelModel.ts#L547). If it's set, it should have been forwarded to the panel migrator [here](https://github.com/grafana/grafana/blob/academo/add-set-migration-handler-custom-type/public/app/features/dashboard/state/PanelModel.ts#L471), no? The same for the other properties.

[the remark this answers is not in the document]
Comment (academo) [reply packages/grafana-data/src/types/panel.ts:138]: @andresmgot I completely understand your question here and I agree that it doesn't make sense, but in reality, not all of the attributes in the panel model are passed to the setMigrationHandler. This is a breaking change that was introduced in scenes.

The following capture is the "real" `panel` object `setMigrationHandler` gets in a react-to-react panel plugin migration. In an angular-to-react migration it contains the `targets` that's why it is marked as optional here. 

image

[the remark this answers is not in the document]
Comment (andresmgot) [reply packages/grafana-data/src/types/panel.ts:138]: 👍  understood

**Review by andresmgot — APPROVED** [review]

**Pull request merged**